src/classes/PMT_PaymentCreator.cls

```diff
@@ -75,6 +75,56 @@ public with sharing class PMT_PaymentCreator {
         }
         set;
     }
+    
+    /*******************************************************************************************************
+    * @description set of Opportunity Recordtype Id's to not create Payments for.
+    */ 
+    private static set<id> setOppRecordTypeIdNoPayment {
+        get {
+            if (setOppRecordTypeIdNoPayment == null) {
+                string strProp = UTIL_CustomSettingsFacade.getContactsSettings().Opp_RecTypes_Excluded_for_Payments__c;
+                if (strProp != null) {
+                    set<string> setStr = new set<string>(strProp.split(';'));    
+                    setOppRecordTypeIdNoPayment = UTIL_RecordTypes.GetRecordTypeIdSet('Opportunity', setStr);
+                } else {
+                    setOppRecordTypeIdNoPayment = new set<Id>();
+                }
+            }
+            return setOppRecordTypeIdNoPayment;
+        }
+        set;
+    }
+
+    /*******************************************************************************************************
+    * @description set of Opportunity Type values to not create Payments for.
+    */ 
+    private static set<string> setOppTypeNoPayment {
+        get {
+            if (setOppTypeNoPayment == null) {
+                string strProp = UTIL_CustomSettingsFacade.getContactsSettings().Opp_Types_Excluded_for_Payments__c;
+                if (strProp != null) {
+                    setOppTypeNoPayment = new set<string>(strProp.split(';'));    
+                } else {
+                    setOppTypeNoPayment = new set<string>();
+                }
+            }
+            return setOppTypeNoPayment;
+        }
+        set;
+    }
+
+    /*******************************************************************************************************
+    * @description specifies whether Payments should be excluded for the Opportunity
+    * @param opp the Opportunity to test.
+    * @return boolean true if Payments excluded for the Opportunity
+    */ 
+    private static boolean isNoPaymentForOpp(Opportunity opp) {
```

Comment:
At what line specifically do you need it outside of this routine?

On Tue, Nov 24, 2015 at 12:18 PM, David Habib <notifications@github.com>
wrote:

> In src/classes/PMT_PaymentCreator.cls
> <https://github.com/SalesforceFoundation/Cumulus/pull/1827#discussion_r45788884>
> :
>
> > +                    setOppTypeNoPayment = new set<string>(strProp.split(';'));
> > +                } else {
> > +                    setOppTypeNoPayment = new set<string>();
> > +                }
> > +            }
> > +            return setOppTypeNoPayment;
> > +        }
> > +        set;
> > +    }
> > +
> > +    /*******************************************************************************************************
> > +    * @description specifies whether Payments should be excluded for the Opportunity
> > +    * @param opp the Opportunity to test.
> > +    * @return boolean true if Payments excluded for the Opportunity
> > +    */
> > +    private static boolean isNoPaymentForOpp(Opportunity opp) {
>
> the point is even if you add it within this routine, you still need to
> keep testing it outside of this routine like the current code does.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SalesforceFoundation/Cumulus/pull/1827/files#r45788884>
> .
>